### DISCUSSION TITLE
Template routes don't work properly with static segments

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#3005] [Feature Requests]
**Description of the problem**
If you have a static segment, routes don't always work like they should.

**How To Reproduce**
Create template named 'test' put 'I am test' on it.

Make a route to that template: `some/fake/thing/{page:pagination}` and require segments=no. 

some/fake/thing-test/
some/fake/thing-test/P4

Those should both work, but some/fake/thing/ does not.  It just shows the default- aka index page for the template group.

**Additional context**

For the short term, you can make it work by turning the static segment into a variable via regex:

`some/fake/{seg_3:regex[(thing\-test)]}/{page:pagination}`

**Comment by intoeetive** [comment]
I have spent some time digging this and I can confirm the problem.

It seems to be issue if at least one static segment is used.
So if you have regex like `test/{section:alpha_dash}/{page:pagination}` - `test` URL will not resolve the correct template.

EE Template Routes needs more improvements in general (e.g. you can't have multiple routes to same template) we might need to address this in dedicated sprint.
